src/main/webapp/app/entities/attachment/attachment.service.ts

```diff
@@ -50,6 +50,28 @@ export class AttachmentService {
         return this.http.delete<any>(`${this.resourceUrl}/${id}`, { observe: 'response' });
     }
 
+    /**
+     * Requests an access token from the server to download the file. If the access token was generated successfully, the file is then downloaded.
+     *
+     * @param downloadUrl url that is stored in the attachment model
+     */
+    downloadAttachment(downloadUrl: string) {
+        const fileName = downloadUrl.split('/').slice(-1)[0];
+        return this.http
+            .get('api/files/attachments/access-token/' + fileName, { observe: 'response', responseType: 'text' })
+            .switchMap(result => this.http.get(`${downloadUrl}?access_token=${result.body}`, { observe: 'response', responseType: 'blob' }))
+            .do(response => {
+                const blob = new Blob([response.body!], { type: response.headers.get('content-type')! });
+                const url = window.URL.createObjectURL(blob);
+                const link = document.createElement('a');
+                link.setAttribute('href', url);
```

Comment:
Can we not just open a link to `${downloadUrl}?access_token=${result.body}`?

Reply:
not really. browser mostly bock async links. so this is a better way to go

Reply:
As far as I remember, we had some problems with the blob approach before. Are we sure that those issues are resolved now? Because otherwise many students will again report that they won't be able to download the slides.

Reply:
@maximilianmeier ?

Reply:
I'll deploy it one more time to be sure. But at least during testing there were no problems with several browsers

Reply:
i tested it again on Safari, Firefox, Chrome with and without adblock, and Brave. All browsers worked as intended and dowloaded the file.

Reply:
@maximilianmeier I still don't understand why this should work better now as before? 

What have you changed?

The blob approach would also work with the normal jwt token of the user, wouldn't it?

Reply:
i can't really remember what the problem was exactly. I remembered another solution that does not use any blobs. 

i refactored the code again. now it opens a new tab with the access token in it. this should prevent any issues with blobs in the future.

Reply:
@krusche tested everything on the test server again. everything is working